src/components/TodoList/TodoList.tsx

```diff
@@ -1 +1,120 @@
-export const TodoList = () => {};
+import { useState, useEffect } from 'react';
+import { TodoInfo } from '../TodoInfo';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import usersFromServer from '../../api/users';
+
+type Props = {
+  todos: Todo[],
+};
+
+function getUser(id:number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === id);
+
+  return foundUser || null;
+}
+
+function createNewTodo(
+  todos: Todo[],
+  id: number,
+  title: string,
+) {
+  const foundUser = getUser(id);
+  const largestId = todos.reduce((acc, todo) => {
+    return acc < todo.id ? todo.id : acc;
+  }, 0);
+
+  const newId = largestId + 1;
+
+  return {
+    userId: id,
+    id: newId,
+    title,
+    completed: false,
+    user: foundUser,
+  };
+}
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  const [id, setId] = useState(0);
+  const [newTitle, setNewTitle] = useState('');
+  const [submited, setSubmited] = useState(false);
+  const [todoList, setTodoList] = useState(todos);
+
+  const [idError, setIdError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+
+  const handleSubmit = (submitEvent: React.SyntheticEvent) => {
+    submitEvent.preventDefault();
+    setIdError(!id && true);
+    setTitleError(!newTitle && true);
+    setSubmited(current => (!current));
+  };
+
+  useEffect(() => {
+    setIdError(id === null);
+  }, [id]);
+
+  useEffect(() => {
+    setTitleError(newTitle === null);
+  }, [newTitle]);
```

Comment:
You don't need to use useEffect here. Just move the logic into handleSubmit

Reply:
Well, this one was needed to follow working exemple.

Reply:
![Screenshot 2023-06-04 145813](https://github.com/mate-academy/react_add-todo-form/assets/97535253/4e343f40-fcec-4377-b82c-2cf62e626442)

Reply:
and its needed also to pass the tests